pymchelper/readers/shieldhit.py

```diff
@@ -9,9 +9,306 @@
+        return _geotyp_units.get(geotyp, _default_units)
+
+    @staticmethod
+    def get_detector_unit(detector_type, geotyp):
+        """
+        TODO
+        :param detector_type:
+        :param geotyp:
+        :return:
+        """
+        if geotyp == SHGeoType.zone:
+            dose_units = ("MeV/primary", "Dose*volume")
+            dose_gy_units = ("J", "Dose*volume")
+            alanine_units = ("MeV/primary", "Alanine RE*Dose*volume")
+            alanine_gy_units = ("J", "Alanine RE*Dose*volume")
+        else:
+            dose_units = (" MeV/g/primary", "Dose")
+            dose_gy_units = ("Gy", "Dose")
+            alanine_units = ("MeV/g/primary", "Alanine RE*Dose")
+            alanine_gy_units = ("Gy", "Alanine RE*Dose")
+
+        _detector_units = {
+            SHDetType.unknown: ("(nil)", "None"),
+            SHDetType.energy: ("MeV/primary", "Energy"),
+            SHDetType.fluence: (" cm^-2/primary", "Fluence"),
+            SHDetType.crossflu: (" cm^-2/primary", "Planar fluence"),
+            SHDetType.letflu: (" MeV/cm", "LET fluence"),
+            SHDetType.dose: dose_units,
+            SHDetType.dose_gy: dose_gy_units,
+            SHDetType.dlet: ("keV/um", "dose-averaged LET"),
+            SHDetType.tlet: ("keV/um", "track-averaged LET"),
+            SHDetType.avg_energy: ("MeV", "Average energy"),
+            SHDetType.avg_beta: ("(dimensionless)", "Average beta"),
+            SHDetType.material: ("(nil)", "Material number"),
+            SHDetType.alanine: alanine_units,
+            SHDetType.alanine_gy: alanine_gy_units,
+            SHDetType.counter: ("/primary", "Particle counter"),
+            SHDetType.pet: ("/primary", "PET isotopes"),
+            SHDetType.dletg: ("keV/um", "dose-averaged LET"),
+            SHDetType.tletg: ("keV/um", "track-averaged LET"),
+            SHDetType.zone: ("(dimensionless)", "Zone#"),
+            SHDetType.medium: ("(dimensionless)", "Medium#"),
+            SHDetType.rho: ("g/cm^3", "Density"),
+        }
+        return _detector_units.get(detector_type, ("(nil)", "(nil)"))
+
+
+class _SHBinaryReader0p6:
+    """
+    Binary format reader from version >= 0.6
+    """
+    def __init__(self, filename):
+        self.filename = filename
+
+    def read(self, detector, nscale=1):
+        with open(self.filename, "rb") as f:
+            d1 = np.dtype([('magic', 'S6'),
+                           ('end', 'S2'),
+                           ('vstr', 'S16')])
+
+            x = np.fromfile(f, dtype=d1, count=1)  # read the data into numpy
+
+            print(x['magic'][0])
+            print(x['end'][0])
+            print(x['vstr'][0])
+
+            while(f):
+                pl_id, _pl_type, _pl_len, pl = self.get_token(f)
+                if pl_id is None:
+                    break
+
+                # TODO: some clever mapping could be done here surely
+
+                if pl_id == SHBDOTagID.shversion:
```

Comment:
what about following mapping:
```
mapping = { SHBDOTagID.shversion : "shversion",
  SHBDOTagID.filedate : "filedate",
  (...)
  SHBDOTagID.det_xyz_start: ("xmin", "ymin", "zmin")
}

```
Then unpacking can be done for most of items like this:
```
for field in enumerate(mapping[pl_id] :
    if str(field).isalnum():
        setattr( detector, field, pl)
    else:
        for fieldname, payload_value in zip(field, pl) :
             setattr( detector, fieldname, payload_value)
```